src/fsipy/automatedPostprocessing/postprocessing_h5py/spectrograms.py

```diff
@@ -0,0 +1,747 @@
+                             "Re_b are 0, the plot won't be against Reynolds number.")
+    parser.add_argument('--interface-only', action='store_true',
+                        help="Generate spectrogram only for the fluid-solid interface. If present, interface-only "
+                             "spectrogram will be generated; otherwise, the volumetric spectrogram will include all "
+                             "fluid in the sac or all nodes through the wall.")
+    parser.add_argument('--component', type=str, default="mag",
+                        help="Component of the data to visualize. Choose 'x', 'y', 'z', or 'mag' (magnitude).")
+    parser.add_argument('--sampling-method', type=str, default="RandomPoint",
+                        help="Sampling method for spectrogram generation. Choose from 'RandomPoint' (random nodes), "
+                             "'SinglePoint' (single point specified by '--point-id'), or 'Spatial' (ensures uniform "
+                             "spatial sampling, e.g., in the case of fluid boundary layer, the sampling will not bias "
+                             "towards the boundary layer).")
+    parser.add_argument('--n-samples', type=int, default=10000,
+                        help="Number of samples to generate spectrogram data (ignored for SinglePoint sampling).")
+    parser.add_argument('--point-id', type=int, default=-1000000,
+                        help="Point ID for SinglePoint sampling. Ignored for other sampling methods.")
+    parser.add_argument('--overlap-frac', type=float, default=0.75,
+                        help="Fraction of overlap between adjacent windows.")
+    parser.add_argument('--window', type=str, default="blackmanharris",
+                        help="Window function to be used for spectrogram computation. "
+                             "Choose from window types available at "
+                             "https://docs.scipy.org/doc/scipy/reference/signal.windows.html. "
+                             "Default is 'blackmanharris'.")
+    parser.add_argument('--num-windows-per-sec', type=int, default=4,
+                        help="Number of windows per second for spectrogram computation.")
+    parser.add_argument('--thresh-val', type=int, default=None,
+                        help="Threshold value for the spectrogram. Default is determined based on the 'quantity' "
+                             "argument: if 'd', default is -42; if 'v', default is -20; if 'p', default is -5; if "
+                             "'wss', default is -18.")
+    parser.add_argument('--max-plot', type=int, default=None,
+                        help="Maximum value for plotting the spectrogram. Default is determined based on the "
+                             "'quantity' argument: if 'd', default is -30; if 'v', default is -7; if 'p', default is "
+                             "5; if 'wss', default is 0.")
```

Comment:
`--thresh-val` and  `--max-plot` defines minimum and maximum value of color range when plotting the spectrogram. I think it would be nice to rename those two so that it is more clear what we mean by these two names. For example, `--min-color-value` `--max-color-value`. They are `vmin` and `vmax` in matplotlib document, respectively. https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.pcolormesh.html